src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/SqlSer.cs

```diff
@@ -50,16 +52,21 @@ internal static void Serialize(Stream s, object instance)
         //
         // Use a per-thread cache, so that there are no synchronization
         // issues when accessing cache entries from multiple threads.
-        [ThreadStatic]
-        private static Hashtable s_types2Serializers;
+        private static ConcurrentDictionary<Type, Serializer> s_types2Serializers;
 
         private static Serializer GetSerializer(Type t)
         {
+            Serializer s = null;
```

Comment:
NIT: Unnecessary assignment.
```suggestion
            Serializer s;
```